### PR TITLE
Advanced Search Table & Misc Updates

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#7325

*Description of changes:*

* Implemented the advanced search query.

* Updated Search results table.

* Updated property names to match api query params.

* Updated some form validations and resets.

*Notes:*

* Date range is currently untested due to the api not supporting yet, but it is a wip. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "name-request",
-  "version": "2.1.5",
+  "version": "2.1.6",
```
Comment (severinbeauvais) [inline package.json:3]: This is OK, or we could consider it a "significant change" and set this to 2.2.0 instead. Thoughts?

Comment (cameron-freshworks) [reply package.json:3]: It's a full feature for sure, i think `2.2.0` makes sense!

File: src/components/advanced-search/advanced-search-form.vue [hunk 2]
```diff
@@ -158,6 +160,15 @@ export default class AdvancedSearchForm extends Vue {
     if (!this.applicantFirstName) this.applicantLastNameRules = []
   }
 
+  /** Clear form. */
+  @Watch('formReset')
+  private clearForm (): void {
```
Comment (severinbeauvais) [inline src/components/advanced-search/advanced-search-form.vue:165]: Would you want to do this only when `formReset` becomes True? (ie, `if (val) then ...`)

File: src/components/advanced-search/advanced-search-retrieve.vue [hunk 3]
```diff
@@ -147,6 +150,14 @@ export default class AdvancedSearchRetrieve extends Vue {
     }
   }
 
+  /** Clear form. */
+  @Watch('formReset')
+  private clearForm (): void {
```
Comment (severinbeauvais) [inline src/components/advanced-search/advanced-search-retrieve.vue:155]: Same comment as above.

File: src/components/advanced-search/advanced-search-form.vue (hunk 2 above)
Comment (cameron-freshworks) [reply src/components/advanced-search/advanced-search-form.vue:165]: Not exactly. 
In this case i want to do this when the dialog closes or opens, so when the dialog state changes.

Comment (severinbeauvais) [reply src/components/advanced-search/advanced-search-form.vue:165]: Is `formReset` tied to the dialog opening or closing?

Comment (severinbeauvais) [reply src/components/advanced-search/advanced-search-form.vue:165]: Oh, it is...
```
:formReset="dialog"
```

Comment (severinbeauvais) [reply src/components/advanced-search/advanced-search-form.vue:165]: Then why not call the prop something more topical?

As I reads now, I would expect this property to reset the form only when it's set to True.

Comment (cameron-freshworks) [reply src/components/advanced-search/advanced-search-form.vue:165]: Sure

Comment (severinbeauvais) [reply src/components/advanced-search/advanced-search-form.vue:165]: A name that makes more sense to me is "resetOnToggle" (but I don't like this name either 😛 ).

Comment (cameron-freshworks) [reply src/components/advanced-search/advanced-search-form.vue:165]: Updated 👍

File: src/components/advanced-search/advanced-search-retrieve.vue (hunk 3 above)
Comment (cameron-freshworks) [reply src/components/advanced-search/advanced-search-retrieve.vue:155]: Updated 👍

File: src/components/advanced-search/advanced-search-form.vue (hunk 2 above)
Comment (severinbeauvais) [reply src/components/advanced-search/advanced-search-form.vue:165]: Where I'm coming from is this...

From the parent's point of view, it's informing the component when `dialog` changes. The prop name should reflect that.

From the component's point of view, what it does with that prop (ie, reset the form) is an internal thing.

But a prop named "formReset" implies that the form should be reset when it becomes True.

Comment (cameron-freshworks) [reply src/components/advanced-search/advanced-search-form.vue:165]: For sure! Updated to be more topical.

**Review by severinbeauvais — APPROVED** [review]
Nice code on the FF implementation.

Looks good.

**Comment by cameron-freshworks** [comment]
> Nice code on the FF implementation.
> 
> Looks good.
Thanks!

and thanks for the good review, as always.

**Review by andrepestana-aot — APPROVED** [review]
Just some comments but looks very good to me!

File: src/enums/advanced-search-tabs.ts [hunk 4]
```diff
@@ -3,4 +3,5 @@ export enum AdvancedSearchTabs {
   ADVANCED_SEACH_TABLE,
   ADVANCED_SEACH_RETRIEVE,
   ADVANCED_SEACH_INVALID,
+  ADVANCED_SEACH_NO_RESULTS
```
[the remark this answers is not in the document]
Comment (cameron-freshworks) [reply src/enums/advanced-search-tabs.ts:6]: 😁 Thank you! Good eye

[the remark this answers is not in the document]
Comment (cameron-freshworks) [reply src/enums/advanced-search-tabs.ts:6]: 4 typos
😆 Shame

File: src/interfaces/models.ts [hunk 5]
```diff
@@ -258,11 +258,28 @@ export interface WaitingAddressSearchI {
...
-  endDate: string
+  compName: string
+  firstName: string
+  lastName: string
+  submittedStartDate: string
+  submittedEndDate: string
+}
+
+// The complete results from advanced nr search
+export interface AdvancedSearchResultsI {
```
[the remark this answers is not in the document]
Comment (cameron-freshworks) [reply src/interfaces/models.ts:269]: I agree! In Name request however, they've used `I` opposed to the IF we used throughout the Entities UI's. 
So in this case, i wanted to keep it consistent.

File: src/enums/advanced-search-tabs.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (cameron-freshworks) [reply src/enums/advanced-search-tabs.ts]: All fixed up!

**Pull request merged**